### PR TITLE
Edits to allow Python 3.5.3 as minimum

### PR DESCRIPTION
**Pull request by eavanvalkenburg**
Just went through this, I created a virtualenv with Python 3.5.3. Just had to make some small adjustments in the tests, nothing in the source. Some of the tests used the new Python 3.6 syntax for strong typing, replaced that with Asserts on the types after create. And one test used a function from the Secrets package that is new in 3.6, replaced that with a direct function when running on Python < 3.6.

Otherwise, all tests work! So raised the version to 1.1.1 and updated the setup.py for the minimum version of Python.

image
         
## This PR closes/fixes the following issues:
[//]: # (If for some reason your pull request does not require a test case you can just mark this box as checked and explain why it does not require a test case.)
 - Closes bunq/sdk_python#108
    - [x] Tested with virtualenv on Python 3.5.3 (current stable Debian pre-installed version of Python)

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-1.1.0
\ No newline at end of file
+1.1.1
```
Comment (OGKevin) [inline VERSION]: Please revert this back. When a new version is released, we will do the version bump.

File: bunq/sdk/client.py [hunk 2]
```diff
@@ -49,7 +49,7 @@ class ApiClient(object):
     HEADER_RESPONSE_ID_LOWER_CASED = 'x-bunq-client-response-id'
 
     # Default header values
-    _USER_AGENT_BUNQ = 'bunq-sdk-python/1.1.0'
+    _USER_AGENT_BUNQ = 'bunq-sdk-python/1.1.1'
```
Comment (OGKevin) [inline bunq/sdk/client.py]: same

File: setup.py [hunk 3]
```diff
@@ -24,7 +24,7 @@
     # Versions should comply with PEP440.  For a discussion on single-sourcing
     # the version across setup.py and the project code, see
     # https://packaging.python.org/en/latest/single_source_version.html
-    version='1.1.0',
+    version='1.1.1',
```
Comment (OGKevin) [inline setup.py]: same

File: tests/model/generated/endpoint/test_monetary_account_bank.py [hunk 4]
```diff
@@ -1,4 +1,10 @@
-from secrets import token_hex
+try:
+    from secrets import token_hex
+except ImportError:
+    from os import urandom
+    def token_hex():
```
Comment (OGKevin) [inline tests/model/generated/endpoint/test_monetary_account_bank.py]: please add a new line before the function.

File: tests/model/generated/endpoint/test_monetary_account_joint.py [hunk 5]
```diff
@@ -37,10 +37,10 @@ def test_monetary_account_joint_parser(self):
         with open(file_path, self._FILE_MODE_READ) as f:
             json_string = f.read()
 
-        joint_account: MonetaryAccountJoint = MonetaryAccountJoint.from_json(
+        joint_account = MonetaryAccountJoint.from_json(
             json_string
         )
-
+        self.assertIsInstance(joint_account, MonetaryAccountJoint)
```
Comment (OGKevin) [inline tests/model/generated/endpoint/test_monetary_account_joint.py]: please add a new line before the assertion. It is better for readability to group assertions together.

File: tests/model/generated/endpoint/test_payment.py [hunk 6]
```diff
@@ -52,16 +52,14 @@ def test_payment_batch(self):
         """
         """
 
-        response_create: endpoint.BunqResponseInt = \
-            endpoint.PaymentBatch.create(
+        response_create = endpoint.PaymentBatch.create(
                 self.__create_payment_list()
             )
-
+        self.assertIsInstance(response_create, endpoint.BunqResponseInt)
```
Comment (OGKevin) [inline tests/model/generated/endpoint/test_payment.py]: same

File: tests/model/generated/endpoint/test_payment.py [hunk 7]
```diff
@@ -52,16 +52,14 @@ def test_payment_batch(self):
...
         self.assertIsNotNone(response_create)
 
-        response_get: endpoint.BunqResponsePaymentBatch = \
-            endpoint.PaymentBatch.get(response_create.value)
-
+        response_get = endpoint.PaymentBatch.get(response_create.value)
+        self.assertIsInstance(response_get, endpoint.BunqResponsePaymentBatch)
```
Comment (OGKevin) [inline tests/model/generated/endpoint/test_payment.py]: same, please move this one line down.

File: tests/model/generated/endpoint/test_payment.py [hunk 8]
```diff
@@ -83,5 +81,6 @@ def __create_payment_list(self) -> List[endpoint.Payment]:
                     self._PAYMENT_DESCRIPTION
                 )
             )
-
+        self.assertIsInstance(all_payment, List)
```
Comment (OGKevin) [inline tests/model/generated/endpoint/test_payment.py]: same, please add a new line before this assertion.

File: .vscode/settings.json [hunk 9]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (OGKevin) [inline .vscode/settings.json]: Please delete this file.

File: setup.py [hunk 10]
```diff
@@ -56,11 +56,12 @@
 
         # Specify the Python versions you support here. In particular, ensure
         # that you indicate whether you support Python 2, Python 3 or both.
+        'Programming Language :: Python :: 3.5.3',
         'Programming Language :: Python :: 3.6',
     ],
 
     # Require Python version equal or higher than Python 3.6.
```
Comment (kojoru) [inline setup.py]: Please fix the comment here

Comment (kojoru) [reply setup.py]:
```suggestion
     # Require Python version equal or higher than specified version
```

File: tests/model/generated/endpoint/test_monetary_account_bank.py [hunk 11]
```diff
@@ -1,4 +1,11 @@
-from secrets import token_hex
+try:
+    from secrets import token_hex
+except ImportError:
+    from os import urandom
+
```
[the remark this answers is not in the document]
Comment (kojoru) [reply tests/model/generated/endpoint/test_monetary_account_bank.py]:
```suggestion

```

File: tests/model/generated/endpoint/test_payment.py [hunk 12]
```diff
@@ -52,16 +52,16 @@ def test_payment_batch(self):
         """
         """
 
-        response_create: endpoint.BunqResponseInt = \
-            endpoint.PaymentBatch.create(
+        response_create = endpoint.PaymentBatch.create(
                 self.__create_payment_list()
             )
-
+
```
[the remark this answers is not in the document]
Comment (kojoru) [reply tests/model/generated/endpoint/test_payment.py]:
```suggestion

```

**Pull request merged**